internal/clients/team2/disaster.go

```diff
@@ -0,0 +1,195 @@
+func GetMinMaxCoordinate(minOrMax bool, coordinate1 shared.Coordinate, coordinate2 shared.Coordinate) shared.Coordinate {
+	if (minOrMax == Min && coordinate1 < coordinate2) || (minOrMax == Max && coordinate1 > coordinate2) {
+		return coordinate1
+	}
+	return coordinate2
+}
+
+// GetMinMaxFloat is the same as GetMinMaxCoordinate but works for floats
+func GetMinMaxFloat(minOrMax bool, value1 float64, value2 float64) float64 {
+	if (minOrMax == Min && value1 < value2) || (minOrMax == Max && value1 > value2) {
+		return value1
+	}
+	return value2
+}
+
+// MakeDisasterPrediction is used to provide our island's prediction on the next disaster
+func (c *client) MakeDisasterPrediction() shared.DisasterPredictionInfo {
+	totalTurns := float64(c.gameState().Turn)
+
+	// Get the location prediction
+	locationPrediction := GetLocationPrediction(c)
+
+	// Get the time until next disaster prediction and confidence
+	sampleMeanX, timeRemainingPrediction := GetTimeRemainingPrediction(c, totalTurns)
+	confidenceTimeRemaining := GetTimeRemainingConfidence(totalTurns, sampleMeanX)
+
+	// Get the magnitude prediction and confidence
+	sampleMeanM, magnitudePrediction := GetMagnitudePrediction(c, totalTurns)
+	confidenceMagnitude := GetMagnitudeConfidence(totalTurns, sampleMeanM)
+
+	// Get the overall confidence in these predictions
+	confidencePrediction := GetConfidencePrediction(confidenceTimeRemaining, confidenceMagnitude)
+
+	// Get trusted islands NOTE: CURRENTLY JUST ALL ISLANDS
+	trustedislands := GetTrustedIslands()
+
+	// Put everything together and return the whole prediction we have made and teams to share with
+	disasterPrediction := shared.DisasterPrediction{
+		CoordinateX: locationPrediction.X,
+		CoordinateY: locationPrediction.Y,
+		Magnitude:   magnitudePrediction,
+		TimeLeft:    timeRemainingPrediction,
+		Confidence:  confidencePrediction,
+	}
+	disasterPredictionInfo := shared.DisasterPredictionInfo{
+		PredictionMade: disasterPrediction,
+		TeamsOfferedTo: trustedislands,
+	}
+	return disasterPredictionInfo
+}
+
+// GetLocationPrediction provides a prediction about the location of the next disaster.
+// The prediction is always the the centre of the archipelago
+func GetLocationPrediction(c *client) CartesianCoordinates {
+	archipelagoGeography := c.gameState().Geography
+	archipelagoCentre := CartesianCoordinates{
+		X: archipelagoGeography.XMin + (archipelagoGeography.XMax-archipelagoGeography.XMin)/2,
+		Y: archipelagoGeography.YMin + (archipelagoGeography.YMax-archipelagoGeography.YMin)/2,
+	}
+	return archipelagoCentre
+}
+
+// GetTimeRemainingPrediction returns a prediction about the time remaining until the next disaster and the sample mean
+// of the RV X. The prediction is 1/sample mean of the Bernoulli RV, minus the turns since the last disaster.
+func GetTimeRemainingPrediction(c *client, totalTurns float64) (float64, int) {
+	totalDisasters := float64(len(c.disasterHistory))
+	sampleMeanX := totalDisasters / totalTurns
```

Comment:
multiple ~vibe~ div0 checks required here

Reply:
totalturns is always larger than 1